catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java

```diff
@@ -0,0 +1,706 @@
+            updateRegistryConfigurations(metacard, false);
+        } catch (IOException | InvalidSyntaxException | ParserException | RuntimeException e) {
+            LOGGER.error("Unable to update registry configurations, metacard still updated");
+        }
+    }
+
+    private void processDelete(Metacard metacard) {
+        try {
+            if (cleanUpOnDelete) {
+                deleteRegistryConfigurations(metacard);
+            }
+        } catch (IOException | InvalidSyntaxException e) {
+            LOGGER.error("Unable to delete registry configurations, metacard still deleted");
+        }
+    }
+
+    /**
+     * Finds all source configurations associated with the registry metacard creates/updates them
+     * with the information in the metacards service bindings. This method will enable/disable
+     * configurations if the right conditions are met but will never delete a configuration other
+     * than for switching a configuration from enabled to disabled or vice-versa
+     *
+     * @param metacard    Registry metacard with new service binding info
+     * @param createEvent Flag indicating if this was a metacard create or update event
+     * @throws IOException
+     * @throws InvalidSyntaxException
+     * @throws ParserException
+     */
+    private void updateRegistryConfigurations(Metacard metacard, boolean createEvent)
+            throws IOException, InvalidSyntaxException, ParserException {
+
+        boolean identityNode =
+                metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_IDENTITY_NODE) != null;
+
+        boolean autoActivateConfigurations = activateConfigurations && !identityNode && (createEvent
+                || !preserveActiveConfigurations);
+
+        List<ServiceBindingType> bindingTypes = registryTypeHelper.getBindingTypes(
+                metacardMarshaller.getRegistryPackageFromMetacard(metacard));
+
+        String registryId = metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                .getValue()
+                .toString();
+
+        String configId = getDeconflictedConfigId(metacard.getTitle(), registryId);
+
+        Map<String, Configuration> fpidToConfigurationMap = getCurrentConfigurations(registryId);
+
+        String bindingTypeToActivate = getBindingTypeToActivate(bindingTypes);
+        String fPidToActivate = bindingTypeToFactoryPidMap.get(bindingTypeToActivate);
+
+        if (autoActivateConfigurations) {
+            activateDeactivateExistingConfiguration(fPidToActivate, fpidToConfigurationMap);
+        }
+
+        for (ServiceBindingType bindingType : bindingTypes) {
+            Map<String, Object> slotMap = this.getServiceBindingProperties(bindingType);
+
+            if (slotMap.get(BINDING_TYPE) == null) {
+                continue;
+            }
+            String factoryPidMask = slotMap.get(BINDING_TYPE)
+                    .toString();
+            String factoryPid = bindingTypeToFactoryPidMap.get(factoryPidMask);
```

Comment:
What will happen if `bindingTypeToFactoryPidMap` has changed since we called it at L217 or while we're inside this loop?